src/rules/dynamic-import-chunkname.js

```diff
@@ -0,0 +1,71 @@
+import docsUrl from '../docsUrl'
+
+module.exports = {
+  meta: {
+    docs: {
+      url: docsUrl('dynamic-import-chunkname'),
+    },
+    schema: [{
+      type: 'object',
+      properties: {
+        importFunctions: {
+          type: 'array',
+          uniqueItems: true,
+          items: {
+            type: 'string',
+          },
+        },
+        webpackChunknameFormat: {
+          type: 'string',
+        },
+      },
+    }],
+  },
+
+  create: function (context) {
+    const config = context.options[0]
+    const { importFunctions = [] } = config || {}
+    const { webpackChunknameFormat = '[0-9a-zA-Z-_/.]+' } = config || {}
+
+    const commentFormat = ` webpackChunkName: "${webpackChunknameFormat}" `
+    const commentRegex = new RegExp(commentFormat)
+
+    return {
+      [`CallExpression[callee.type="Import"],CallExpression[callee.name]`](node) {
```

Comment:
@ljharb, let me know if you know of a better way to write these selectors now that we're accepting an array of import functions instead of just a string

Reply:
hmm - i don't think it has to be a blocker or anything, but you *could*, perhaps, dynamically construct the method name like so:
```js
const selector = ['CallExpression[callee.type="Import"]'].concat(importFunctions.map(func => `CallExpression[callee.name="${func}"]`).join(',');

return {
  [selector](node) {
```

but at this point I'm not sure if the readability hit is worth the performance optimization. What do you think?

Reply:
I think this may be what's failing the Travis build for eslint version 2

Reply:
oh, snap - good call. if eslint 2 doesn't have full AST selectors, then we should go back to your original code that just checks `CallExpression` and does the filtering inside the method body :-(